src/libs/error.test.ts

```diff
@@ -0,0 +1,135 @@
+import { sessionTimedOutErrorMessage } from 'src/auth/auth-errors';
+import { reportError as reportErrorFallback, withErrorReporter } from 'src/libs/error';
+import { notify } from 'src/libs/notifications';
+import { mockNotifications } from 'src/testing/test-utils';
+
+jest.mock('src/libs/notifications');
+
+describe('report', () => {
+  it('calls notify and console.error', async () => {
+    // Arrange
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+    const { reportError } = withErrorReporter(mockNotifications);
+
+    let outerThrow: unknown | null = null;
+
+    // Act
+    try {
+      await reportError('Things went BOOM', new Error('BOOM!'));
+    } catch (e) {
+      outerThrow = e;
+    }
```

Comment:
It looks like this could be simplified to:

```suggestion
    // Act
    await reportError('Things went BOOM', new Error('BOOM!'));
```

Similar comment for mosts tests in this module.

Reply:
not sure I follow... tests cannot throw unhandled exceptions... (considered a test fail) ... hence the try/catch pattern.

Reply:
Exactly, the test should fail if `reportError` throws an exception. That's what the try/catch and `expect(outerThrow).toEqual(null);` currently does, but it uses a lot more code to do it.

Reply:
ah, I missed the toEqual(null) reversal... must have forgotten to go back and clean these up... good catch!